src/ert/config/queue_config.py

```diff
@@ -111,6 +120,23 @@ def create_local_copy(self) -> QueueConfig:
         )
 
 
+def _check_for_overwritten_queue_system_options(
+    selected_queue_system: QueueSystem, queue_system_options: List[Tuple[str, str]]
+) -> None:
+    temp_dict: Dict[str, List[str]] = defaultdict(list)
+    for option_string in queue_system_options:
```

Comment:
why are we checking if instance is tuple if we are expecting `List[Tuple[str, str]]`
If the type hint is not what we expect, then we either explicitly hint unstructured data type or data needs to sturctured before referencing this function.
then the function can be simple
```python
for option_name, option_value in queue_system_options:
    temp_dict.setdefault(option_name, []).append(option_value)
```

Reply:
I had to modify and complicate this part because I discovered that the option_value might be left unspecified in the config file in which case there is only one value to unpack.